src/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -177,122 +172,57 @@ Collections.IEnumerator Collections.IEnumerable.GetEnumerator()
-            }
-        }
 
-        private void WriteNextContentAsync(Task task)
-        {
-            if (task.IsFaulted)
-            {
-                HandleAsyncException("WriteNextContentAsync", task.Exception.GetBaseException());
-                return;
-            }
-
-            try
-            {
-                HttpContent content = _nestedContent[_nextContentIndex];
-                _nextContentIndex++; // Next call will operate on the next content object.
-
-                content.CopyToAsync(_outputStream)
-                    .ContinueWithStandard(WriteNextContentHeadersAsync);
-            }
-            catch (Exception ex)
-            {
-                HandleAsyncException("WriteNextContentAsync", ex);
-            }
-        }
-
-        // Final step, write the footer boundary.
-        private void WriteTerminatingBoundaryAsync()
-        {
-            try
-            {
-                EncodeStringToStreamAsync(_outputStream, CrLf + "--" + _boundary + "--" + CrLf)
-                    .ContinueWithStandard(task =>
+                    // Add headers
+                    foreach (KeyValuePair<string, IEnumerable<string>> headerPair in content.Headers)
                     {
-                        if (task.IsFaulted)
+                        output.Append(headerPair.Key);
+                        output.Append(": ");
+                        string delim = string.Empty;
+                        foreach (string value in headerPair.Value)
                         {
-                            HandleAsyncException("WriteTerminatingBoundaryAsync", task.Exception.GetBaseException());
-                            return;
+                            output.Append(delim);
+                            output.Append(value);
+                            delim = ", ";
                         }
+                        output.Append(CrLf);
+                    }
+                    output.Append(CrLf); // Extra CRLF to end headers (even if there are no headers).
+
+                    // Write divider, headers, and content
+                    await EncodeStringToStreamAsync(stream, output.ToString()).ConfigureAwait(false);
+                    await content.CopyToAsync(stream).ConfigureAwait(false);
+                }
 
-                        TaskCompletionSource<object> lastTcs = CleanupAsync();
-                        lastTcs.TrySetResult(null); // This was the final operation.
-                    });
+                // Write footer boundary
+                await EncodeStringToStreamAsync(stream, CrLf + "--" + _boundary + "--" + CrLf).ConfigureAwait(false);
             }
             catch (Exception ex)
             {
-                HandleAsyncException("WriteTerminatingBoundaryAsync", ex);
+                if (NetEventSource.Log.IsEnabled()) NetEventSource.Exception(NetEventSource.ComponentType.Http, this, nameof(SerializeToStreamAsync), ex);
```

Comment:
I can add them, but FYI this is just moved from here: 
https://github.com/dotnet/corefx/pull/7379/files/1102890d828a46a865b32b09865b4527e0a93fa3#diff-167318fdd23c477b7592fbb7cd2e425bL317

(Also note that these events don't actually work, due to https://github.com/dotnet/corefx/issues/5755.)